src/GuiBase/Timeline/TimelineFrameSelector.h

```diff
@@ -0,0 +1,326 @@
+     */
+    void endChanged( Scalar time );
+
+    /**
+     * Emitted when the playzone duration has been changed though
+     * an internal call to onChangeDuration().
+     */
+    void durationChanged( Scalar time );
+
+    /**
+     * Emitted when the playzone current time has been changed though
+     *  - the user changing the current time of the Timeline's playzone,
+     *  - the user offsetting a set of KeyFrames,
+     *  - the user deleting an entire timezone.
+     */
+    void cursorChanged( Scalar time );
+
+    /**
+     * Emitted when a KeyFrame has been added through
+     * an internal call to onAddingKeyFrame().
+     */
+    void keyFrameAdded( Scalar time );
+
+    /**
+     * Emitted when a KeyFrame has been deleted through
+     *  - the user deleting a single KeyFrame.
+     *  - the user deleting an entire timezone containing a KeyFrame.
+     * \note Emitted for each suppressed KeyFrame.
+     */
+    void keyFrameDeleted( size_t i );
+
+    /**
+     * Emitted when a KeyFrame has been added through
+     * an internal call to onAddingKeyFrame().
+     */
+    void keyFrameChanged( size_t i );
+
+    /**
+     * Emitted when a KeyFrame has been added through
+     * an internal call to onMoveKeyFrame().
+     */
+    void keyFrameMoved( size_t i, Scalar time1 );
+
+    /**
+     * Emitted when a KeyFrame has been added through
+     *  - the user offsetting a set of keyframes.
+     *  - the user deleting an entire timezone preceding KeyFrames.
+     */
+    void keyFramesMoved( size_t first, Scalar offset );
+
+  private slots:
+    /**
+     * Slides the slider for the start point of the Timeline's playzone, and
+     * resizes the playzone UI.
+     * Also change its color for while it is sliding.
+     * \note Internally calls onChangeStart().
+     */
+    void onSlideLeftSlider( int deltaX );
+
+    /**
+     * Slides the slider for the end point of the Timeline's playzone, and
+     * resizes the playzone UI.
+     * Also change its color for while it is sliding.
+     * \note Internally calls onChangeEnd().
+     */
+    void onSlideRightSlider( int deltaX );
+
+    /**
+     * Set the cursor to the start point of the Timeline's playzone.
+     * \note Internally calls onChangeCursor().
+     */
+    void onSetCursorToStart();
+
+    /**
+     * Set the cursor to the end point of the Timeline's playzone.
+     * \note Internally calls onChangeCursor().
+     */
+    void onSetCursorToEnd();
+
+    /**
+     * Set the cursor to the previous KeyFrame, if any.
+     * \note Internally calls onChangeCursor().
+     */
+    void onSetCursorToPreviousKeyFrame();
+
+    /**
+     * Set the cursor to the next KeyFrame, if any.
+     * \note Internally calls onChangeCursor().
+     */
+    void onSetCursorToNextKeyFrame();
+
+  protected:
+    virtual void paintEvent( QPaintEvent* event ) override;
+
+    void mousePressEvent( QMouseEvent* event ) override;
```

Comment:
Is there u user manual showing which button/key combo does what ? Not rellay easy to understand how to use the timelineselector if one has to read the code.

Reply:
That's on the HelpDialog sheet, which can be opened from the `?` button on the timeline ui